### PR TITLE
FIRDynamicLink: preserve all parameters passed to initializer

### PR DESCRIPTION
**Pull request by maksymmalyhin**
This is a proposed fix for #2462 

TODO: Tests

### CODE REVIEW

**Comment by maksymmalyhin** [comment]
@dmandar @mh6o Please see the issue details here #2462. Feel free to ask me if you have any questions.

File: Firebase/DynamicLinks/FIRDynamicLink.m [hunk 1]
```diff
@@ -32,33 +32,70 @@ - (instancetype)initWithParametersDictionary:(NSDictionary *)parameters {
...
+                              forKey:kFIRDLParameterMatchType];
+}
+
+- (void)setMatchMessage:(NSString *)matchMessage {
+  _matchMessage = [matchMessage copy];
+  [self setParametersDictionaryValue:_matchMessage forKey:kFIRDLParameterMatchMessage];
+}
+
+- (void)setParametersDictionaryValue:(id)value forKey:(NSString *)key {
+  NSMutableDictionary *parametersDictionary = [self.parametersDictionary mutableCopy];
```
Comment (dmandar) [inline Firebase/DynamicLinks/FIRDynamicLink.m]: NSMutableDictionary <NSString *, id> *parametersDictionary

**Comment by harrydema** [comment]
@maksymmalyhin When will this ticket be merged?

Thanks!

**Comment by maksymmalyhin** [comment]
@harrydema Just merged it. The fix should be available next release.

**Comment by damienix** [comment]
Thanks for the swift patch @maksymmalyhin  👍

Fingers crossed till testing this on production ✊ 😃

**Comment by mustahsin** [comment]
Hi. Is the fixation merged and updated in latest sdk? I have installed firebase dynamic link through pod (version: 3.4.3). still didn't get UTM parameters for dynamic_link_first_open event in Conversion.

**Comment by paulb777** [comment]
@mustahsin This fix has been in the last two releases.  You can confirm by checking to see that the source changes in this PR are in the sources in your `Pods` directory.

**Comment by mustahsin** [comment]
yes @paulb777 . I can see the changes added in my project. But in Firebase Dashboard Conversion I am not getting those utm source and campaign parameters. Its logging as direct. I can see those attributes in Debugview but not in Conversion.

Screen Shot 2019-04-02 at 11 17 53 AM

For android sdk source parameters are logged.

Screen Shot 2019-04-02 at 11 21 45 AM

**Comment by MMS129** [comment]
hi. I am still not getting UTM parameters in Conversions for dynamic link app open and first open. I can see the logged attributes for campaign, source in DebugView and StreamView. But not in Conversions. 

Screenshot 2019-05-10 at 4 22 50 PM

Its showing as Direct.

**Pull request merged**